pkg/piecestore/psserver/readerwriter.go

```diff
@@ -114,3 +122,13 @@ func (s *StreamReader) Read(b []byte) (int, error) {
 
 	return n, nil
 }
+
+// approved returns true if a node ID exists in a list of approved node IDs
+func approved(id storj.NodeID, list []storj.NodeID) bool {
```

Comment:
maybe move this to the server file and use a server receiver so you can use server.whitelist rather than having to pass in a list of nodeids ie `func (s *Server) approved(id storj.NodeID) bool {...}`